### PR TITLE
Fixes bin/trace so it doesn't dirty the repo when autogen runs

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

File: bin/trace [hunk 1]
```diff
@@ -207,7 +207,7 @@ $Source = "";
 # Maximum length of H5TRACE macro line
 # If the ColumnLimit in .clang-format is changed, this value will need to be updated
 #
-my $max_trace_macro_line_len = 110;
+my $max_trace_macro_line_len = 109;
```
Comment (derobins) [inline bin/trace]: This line length keeps bin/trace and bin/format_source from fighting over long TRACE macros

File: bin/trace [hunk 2]
```diff
@@ -470,7 +470,7 @@ for $file (@ARGV) {
   $file_args = 0;
 
   # Ignore some files that do not need tracing macros
-  unless ($file eq "H5FDmulti.c" or $file eq "src/H5FDmulti.c" or $file eq "H5FDstdio.c" or $file eq "src/H5FDstdio.c") {
+  unless ($file eq "H5FDmulti.c" or $file eq "src/H5FDmulti.c" or $file eq "H5FDstdio.c" or $file eq "src/H5FDstdio.c" or $file eq "src/H5TS.c") {
```
Comment (derobins) [inline bin/trace:473]: Nothing in H5TS.c has a TRACE macro and this is easier than screwing with bin/trace to get it to stop messing with the file.

**Review by byrnHDF — APPROVED** [review]

**Comment by derobins** [comment]
Why is this failing the format checks?

File: bin/trace (hunk 1 above)
Comment (soumagne) [reply bin/trace]: I feel like there's a +1/-1 that's not respected somewhere then?

**Comment by derobins** [comment]
ARGH. NOW IT'S FIGHTING OVER ANOTHER FILE.

File: bin/trace (hunk 1 above)
Comment (derobins) [reply bin/trace]: Right, but I'm unexcited about digging through a rat's nest of Perl to figure out where it's going wrong. I'm just going to turn the formatter off for that one line.

**Review by lrknox — APPROVED** [review]

File: bin/trace (hunk 2 above)
Comment (qkoziol) [reply bin/trace:473]: This is not a good solution.   Those other files are the "public" examples and H5TS.c is not.

**Review by qkoziol — CHANGES_REQUESTED** [review]
I don't think this is the correct direction to go.   The bin/trace script needs to be updated, not hacks in other files.

**Comment by derobins** [comment]
@qkoziol The bin/trace issues are due to your checkins, so if you want to do a better job at fixing the problems, have at it. I'm just tired of having to engineer around this nonsense. autogen.sh should NEVER dirty the repo on a fresh checkout. Allowing that is insanity.

**Comment by derobins** [comment]
Between this and all the bad format decisions clang-format makes, I'm of a mind to get rid of the formatter altogether.

File: bin/trace (hunk 2 above)
Comment (derobins) [reply bin/trace:473]: H5TS has no calls that require the TRACE macro and I have zero interest in debugging the trace script to fix its processing.

**Comment by qkoziol** [comment]
> @qkoziol The bin/trace issues are due to your checkins, so if you want to do a better job at fixing the problems, have at it. I'm just tired of having to engineer around this nonsense. autogen.sh should NEVER dirty the repo on a fresh checkout. Allowing that is insanity.

Yeah, I can look at bin/trace and try to get it to stop arguing with clang-format.   Especially since it's the component that we have actual control over...

**Comment by lrknox** [comment]
I'm going to merge this to stop continuing issues with other PRs. This PR makes just a few changes which can be reverted when a better solution is available.

**Pull request merged**